types/src/v2/params.rs

```diff
@@ -152,49 +164,30 @@ impl<'a> RpcParams<'a> {
 	}
 
 	/// Attempt to parse all parameters as array or map into type `T`
-	pub fn parse<T>(self) -> Result<T, CallError>
+	pub fn parse<T>(&'a self) -> Result<T, CallError>
 	where
 		T: Deserialize<'a>,
 	{
-		let params = self.0.unwrap_or("null");
+		let params = self.json.as_ref().map(AsRef::as_ref).unwrap_or("null");
```

Comment:
Will the `unwrap_or` allocate here if we're a `Cow::Owned`?

Reply:
No, it's casting it to `Option<&str>` and then unwrapping.